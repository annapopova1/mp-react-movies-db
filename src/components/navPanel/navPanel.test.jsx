import React from 'react';
import NavPanel from './navPanel';

describe('<NavPanel/>', () => {
  test('should render NavPanel component to show Search By panel', () => {
    const links = [
      {
        title: 'TITLE', param: 'title', active: true, handler: jest.fn(),
      },
      {
        title: 'DIRECTOR', param: 'director', active: false, handler: jest.fn(),
      },
    ];
    const btn = { title: 'SEARCH', handler: jest.fn() };
    const navPanel = shallow(<NavPanel direction="left" secondaryBrand="SEARCH BY" links={links} navBtn={btn} />);
    expect(navPanel).toMatchSnapshot();
  });

  test('should render NavPanel component to show Sort By panel', () => {
    const links = [
      {
        title: 'release date', param: 'releaseDate', active: true, handler: jest.fn(),
      },
      {
        title: 'rating', param: 'voteAvg', active: false, handler: jest.fn(),
      },
    ];
    const navPanel = shallow(<NavPanel
      direction="right"
      primaryBrand="7 movies found"
      secondaryBrand="Sort by"
      links={links}
    />);
    expect(navPanel).toMatchSnapshot();
  });
});
