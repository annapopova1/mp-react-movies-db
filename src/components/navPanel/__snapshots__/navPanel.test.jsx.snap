// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavPanel/> should render NavPanel component to show Search By panel 1`] = `
<nav
  className="navbar navbar-light navbar-expand-md"
>
  <div
    className="navbar-nav mr-auto"
  >
    <span
      className="navbar-brand"
    >
      SEARCH BY
    </span>
    <a
      className="nav-link active"
      href="#"
      key="title"
    >
      TITLE
    </a>
    <a
      className="nav-link"
      href="#"
      key="director"
    >
      DIRECTOR
    </a>
  </div>
  <button
    className="btn btn-outline-success"
    onClick={[MockFunction]}
    type="button"
  >
    SEARCH
  </button>
</nav>
`;

exports[`<NavPanel/> should render NavPanel component to show Sort By panel 1`] = `
<nav
  className="navbar navbar-light navbar-expand-md"
>
  <span
    className="navbar-brand"
  >
    7 movies found
  </span>
  <div
    className="navbar-nav ml-auto"
  >
    <span
      className="navbar-brand"
    >
      Sort by
    </span>
    <a
      className="nav-link active"
      href="#"
      key="releaseDate"
    >
      release date
    </a>
    <a
      className="nav-link"
      href="#"
      key="voteAvg"
    >
      rating
    </a>
  </div>
</nav>
`;
